forms/HtmlEditorConfig.php

```diff
@@ -289,35 +289,78 @@ public function removeButtons() {
 			$this->modifyButtons($button, 0, 1);
 		}
 	}
-	
+
 	/**
-	 * Generate the javascript that will set tinyMCE's configuration to that of the current settings of this object
-	 * @return string - the javascript
+	 * Generates the final configuration array which can be used in the TinyMCE editor (converted to JSON).
+	 *
+	 * @return array
 	 */
-	public function generateJS() {
+	public function getConfig() {
+		// Get base settings for this instance and pile on configured internal plugins as well as current button configuration.
 		$config = $this->settings;
-		
-		// plugins
+		$config['plugins'] = implode(',', $this->getInternalPlugins());
+		foreach ($this->buttons as $i=>$buttons) {
+			$config['theme_advanced_buttons'.$i] = implode(',', $buttons);
+		}
+
+		return $config;
+	}
+
+	/**
+	 * Returns an array of internal plugins.
+	 *
+	 * @return array
+	 */
+	public function getInternalPlugins() {
```

Comment:
Can you please make the new functions protected? We want to keep the HTMLEditorConfig as agnostic to the JS implementation as possible, and the "internal" vs "external" definition is specific to TinyMCe.